processing/src/main/java/org/apache/druid/segment/NestedDataColumnMerger.java

```diff
@@ -163,13 +162,8 @@ private GlobalDictionarySortedCollector getSortedIndexFromIncrementalAdapter(
       return null;
     }
     final NestedDataColumnIndexer indexer = (NestedDataColumnIndexer) dim.getIndexer();
-    for (Map.Entry<String, NestedDataColumnIndexer.LiteralFieldIndexer> entry : indexer.fieldIndexers.entrySet()) {
-      // skip adding the field if no types are in the set, meaning only null values have been processed
-      if (!entry.getValue().getTypes().isEmpty()) {
-        mergedFields.put(entry.getKey(), entry.getValue().getTypes());
-      }
-    }
-    return indexer.globalDictionary.getSortedCollector();
+    indexer.mergeFields(mergedFields);
+    return indexer.getSortedCollector();
```

Comment:
This change is just general refactoring and not actually functionally meaningful, right?  Just validating my understanding

Reply:
yep, made writing the test easier and the direct field access was kind of ugly